Features/Context/SubContext/CommonActions.php

```diff
@@ -0,0 +1,363 @@
+
+        $this->attachFile( $path );
+    }
+
+    /**
+     * Attaches a file to a input field en HTML
+     *
+     * @param string $path      file path relative to mink definitions
+     */
+    protected function attachFile( $path )
+    {
+        if ( $this->getMinkParameter( 'files_path' ) )
+        {
+            $fullPath = rtrim( realpath( $this->getMinkParameter( 'files_path' ) ), DIRECTORY_SEPARATOR ).DIRECTORY_SEPARATOR.$path;
+
+            if ( is_file( $fullPath ) )
+            {
+                $fileInput = 'input[type="file"]';
+                $field = $this->getSession()->getPage()->find( 'css', $fileInput );
+
+                if ( null === $field )
+                {
+                    throw new Exception( "File input is not found" );
+                }
+            $field->attachFile( $fullPath );
+            }
+        }
+        else throw new Exception( "File is not found at the given location" );
+    }
+
+    /**
+     * Drops a file to an HTML element with specific text
+     *
+     * @param string $path      file path relative to mink definitions
+     * @param string $selector  selector of the element where to drop the file
+     */
+    protected function dragAndDropFile( $path, $selector )
+    {
+        if ( $this->getMinkParameter( 'files_path' ) )
+        {
+            $fullPath = rtrim( realpath( $this->getMinkParameter( 'files_path' ) ), DIRECTORY_SEPARATOR ).DIRECTORY_SEPARATOR.$path;
+
+            if ( is_file( $fullPath ) )
+            {
+                $fileName = $path;
+                $finfo = finfo_open( FILEINFO_MIME_TYPE );
+                $fileType = finfo_file( $finfo, $fullPath );
+                $contents = file_get_contents( $fullPath );
+                $contentBase64 = base64_encode( $contents );
+
+                $jsArgs = JsHelper::generateFuncArgs( $fileName, $fileType, $contentBase64, $selector );
+                $jsCode = "return BDD.simulateDropEventWithFile( $jsArgs );";
+                $jsFile = $this->execJavascript( $jsCode );
+            }
+        }
+        else throw new Exception( "File is not found at the given location" );
+    }
+
+    /**
+     * Finds an HTML element by class and the text value and return the value of the sibling
+     *
+     * @param string $name
+     */
+    protected function checksImagePresent( $text, $selector )
+    {
+        $jsArgs = JsHelper::generateFuncArgs( $text, $selector );
+        $jsCode = "return BDD.getImgSrc( $jsArgs );";
+        return $this->evalJavascript( $jsCode );
+    }
+
+    /**
+     * Finds an HTML element by class and the text value and return the value of the sibling
+     *
+     * @param string $name
+     */
+    protected function getFileUrl( $text, $selector )
+    {
+        $jsArgs = JsHelper::generateFuncArgs( $text, $selector );
+        $jsCode = "return BDD.getLinkUrl( $jsArgs );";
+        return $this->evalJavascript( $jsCode );
+    }
+
+    /**
+     * Get the current window Node Id
```

Comment:
Yes, this is a workaround because so far there is no information on platformUI about the content no ids, no type, etc. This is a supposed temporary solution for some test cases.